src/enclave/main.cpp

```diff
@@ -46,6 +46,7 @@ extern "C"
     size_t* enclave_version_len,
     StartType start_type,
     ccf::LoggerLevel enclave_log_level,
+    ccf::pal::Platform platform,
```

Comment:
Because the `.so` no longer knows what platform it is at build-time, we need to pass this through from the host. Ugly to add more args here, but we'll nuke the whole thing shortly.